apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -333,9 +333,14 @@ public ErrorCapture captureException(@Nullable Throwable e, @Nullable AbstractSp
 
     @Nullable
     private ErrorCapture captureException(long epochMicros, @Nullable Throwable e, @Nullable AbstractSpan<?> parent, @Nullable ClassLoader initiatingClassLoader) {
-        if (!isRunning()) {
+        if (!isRunning() || e == null) {
             return null;
         }
+
+        while (e != null && WildcardMatcher.anyMatch(coreConfiguration.getUnnestExceptions(), e.getClass().getName()) != null) {
```

Comment:
This is the fix. If we first filter based on `ignore_exceptions` and unnest later, this bug occurs.
Why do you think it should be in the `setException` method?
Changing from single unnesting to loop is only something I added because I thought it's better, it is not the fix.

Reply:
Just that the `WildcardMatcher.isAnyMatch` was in the setException before, I'm wondering why this was moved here. I don't really have any preference to it being here or there, just wondering why you moved it

Reply:
> Just that the WildcardMatcher.isAnyMatch was in the setException before, I'm wondering why this was moved here.

That was the bug 🙂

Reply:
lol got it now